### ISSUE TITLE
mod_bosh: ignores inactivity and terminate session after 60s

### ISSUE THREAD

**Issue by ghost** [esl/MongooseIM#2219]
Hello. I am using default configuration, except that I have changed:
```
{mod_bosh, [{inactivity, 3153600000}]},
```

After acquiring session, authentication, and resource bind I get reply from server:
```xml
<body wait='0' requests='2' hold='1' from='localhost' accept='deflate, gzip' sid='b17a3dde45547a75c368a20d60119c8f639a2a3d' xmpp:restartlogic='true' xmpp:version='1.0' xmlns='http://jabber.org/protocol/httpbind' xmlns:xmpp='urn:xmpp:xbosh' xmlns:stream='http://etherx.jabber.org/streams' inactivity='3153600000' maxpause='120'><stream:features><bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'/><session xmlns='urn:ietf:params:xml:ns:xmpp-session'/><register xmlns='http://jabber.org/features/iq-register'/><sm xmlns='urn:xmpp:sm:3'/></stream:features><iq xmlns='jabber:client' id='1549737780' type='result'><bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'><jid>phpunit@localhost/bot</jid></bind></iq></body>
```

Note inactivity attribute that is what I want. However session is being closed after 60s anyways:
```
18:44:00.963 [debug] Closing session <<"b17a3dde45547a75c368a20d60119c8f639a2a3d">> in 'normal' state. Handlers: [] Pending: []
```

Which is unexpected behavior violating the protocol.

**Comment by mercuriev** [comment]
Session is closed after 60s even when I poll for messages every 2s!

**Comment by mercuriev** [comment]
I found that if I include `<ping xmlns:="urn:xmpp:ping" />` requests in body, then session is not closed.
It seems that underlying XMPP connection is dropped without data and BOSH does not keep it alive.

It is still undesired behavior because client may unexpectedly stop polling and effectively `inactivity` period is 60 seconds.

**Comment by fenek** [comment]
Hi @mercuriev 

This is a known issue we have after upgrading `cowboy` server to the most recent version. Unfortunately most probably it won't be fixed in 3.3.0 release but soon in the future anyway.

**Comment by bartekgorny** [comment]
Looks like in this case mod_bosh was used in a way which is strongly discouraged by XSF - the 'wait' parameter was enforced at 0, which basically means that it wasn't BOSH anymore. BOSH is about long-polling, while here the server would respond immediately and the whole thing becomes a 'regular' polling. 

With proper configuration, the issue is not reproducable. Since the OP is gone from github we can't follow the case further.
